src/Request.hs

```diff
@@ -114,40 +114,40 @@ mkValidator contractInfo@ContractInfo{..} dat lenderTn ctx = validate
     containsNewDatum :: TxOut -> Bool
     containsNewDatum txo = findDatumHash' expectedNewDatum (U.info ctx) == txOutDatumHash txo
 
-    containtsTimeNft :: TxOut -> Bool
-    containtsTimeNft txo = any (\(cs, _tn, n) -> cs == timeNft && n == 1) (flattenValue (txOutValue txo))
-
     doesNotContainAdditionalTokens :: TxOut -> Bool
     doesNotContainAdditionalTokens txo = length ((flattenValue . txOutValue) txo) < 4
 
     txOutValidate :: TxOut -> Bool
     txOutValidate txo = isItToCollateral txo &&
                         containsRequiredCollateralAmount txo &&
                         containsNewDatum txo &&
-                        containtsTimeNft txo &&
                         doesNotContainAdditionalTokens txo
 
     validateTxOuts :: Bool
     validateTxOuts = any txOutValidate (txInfoOutputs $ U.info ctx)
 
+    checkDeadline :: Bool
+    checkDeadline = contains (from (lendDateRdm rdm)) (U.range ctx)
```

Comment:
This check is in the wrong direction, it checks whether the start of the lending period is before the transaction. Lender could do the following:

- for a 1-year lending request, claim that `lendDateRdm` is 11 months and 3 weeks in the past. This passes the check.
- wait one week
- collect interest for the full 1-year lending contract

We think it's necessary to bound the size of `U.range ctx`, typically it's set to 10's of minutes or even hours. This decision depends also on the business case, we can discuss it on Discord or in a call. For now put any constant and name it, so that it can be changed easily.

Reply:
Actually we don't have to bound the range but then this appends the exact start date of the loan contract - it affects the business, so it needs to be considered from the business side.

Reply:
I'm also in favor of skipping lendDateRdm and just using range beginning or end: https://github.com/aadafinance/aada-finance/pull/23#discussion_r934599036